libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKLoginFlowSelectionViewController.m

```diff
@@ -233,20 +231,14 @@ - ( UIImage * _Nonnull )headerBackgroundImage {
 }
 
 - (IBAction)useIDPAction:(id)sender {
-    if ([self.selectionFlowDelegate respondsToSelector:@selector(loginFlowSelectionIDPSelected:)]) {
-        [self.selectionFlowDelegate loginFlowSelectionIDPSelected:self];
+    if ([self.selectionFlowDelegate respondsToSelector:@selector(loginFlowSelectionIDPSelected:options:)]) {
+        [self.selectionFlowDelegate loginFlowSelectionIDPSelected:self options:self.appOptions];
     }
 }
 
 - (IBAction)useLocalAction:(id)sender {
-    if ([self.selectionFlowDelegate respondsToSelector:@selector(loginFlowSelectionLocalLoginSelected:)]) {
-        [self.selectionFlowDelegate loginFlowSelectionLocalLoginSelected:self];
-    }
-}
-
-- (IBAction)cancelAction:(id)sender {
-    if ([self.selectionFlowDelegate respondsToSelector:@selector(loginFlowSelectionCancelSelected:)]) {
-        [self.selectionFlowDelegate loginFlowSelectionCancelSelected:self];
+    if ([self.selectionFlowDelegate respondsToSelector:@selector(loginFlowSelectionLocalLoginSelected:options:)]) {
```

Comment:
Same question: should `loginFlowSelectionLocalLoginSelected` be optional in the protocol? What should happened if it is not defined?